staging/src/k8s.io/kube-aggregator/pkg/controllers/openapi/controller.go

```diff
@@ -139,7 +139,10 @@ func (c *AggregationController) AddAPIService(handler http.Handler, apiService *
 
 // UpdateAPIService updates API Service's info and handler.
 func (c *AggregationController) UpdateAPIService(handler http.Handler, apiService *v1.APIService) {
-	if err := c.openAPIAggregationManager.AddUpdateAPIService(apiService, handler); err != nil {
+	if apiService.Spec.Service == nil {
```

Comment:
this doesn't handle aggregating openapi for custom resources?

Reply:
IIUC, the update here is just to refresh aggregated apiservers by downloading every once in a while. CRDs OpenAPI is refreshed on demand.

Reply:
Yep exactly as @apelisse mentioned. This nil check was part of AddUpdateAPIService, but since we don't call that here anymore, add the check here too.